addons/mail/wizard/mail_compose_message.py

```diff
@@ -701,18 +704,33 @@ def _action_send_mail_mass_mail(self, res_ids, auto_commit=False):
         return mails_sudo
 
     def action_save_as_template(self):
-        """ hit save as template button: current form value will be a new
-            template attached to the current document. """
+        """ hit save as template button: opens a wizard that prompts for the template's subject.
+            `save_as_template` is called when saving the new wizard. """
+
+        wizard = self.copy()
+        wizard.subject = False
+        return {
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'view_id': self.env.ref('mail.email_compose_message_wizard_small_form').id,
+            'name': 'Create a new Template',
+            'res_model': 'mail.compose.message',
+            'context': {'dialog_size': 'medium'},
+            'target': 'new',
+            'res_id': wizard.id,
+        }
+
+    def save_as_template(self):
+        """ saves the current mail composer as a mail template """
         for record in self:
             model = self.env['ir.model']._get(record.model or 'mail.message')
```

Comment:
Would say: time to remove this fallback that makes no sense: we should not be able to invoke the method on a composer that has no model -> raise an Usererror if model is not set (could be done in a separate commit as it is more a technical cleaning). We can also use the "get_id' shortcut like
```suggestion
            if not composer.model or not composer.model in self.env
                raise Usererror(_('Template creation from composer requires a valid model.'))
            model_id = self.env['ir.model']._get_id(record.model)
```